sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -641,14 +641,19 @@ object SQLConf {
     .checkValue(_ > 0, "The value of spark.sql.leafNodeDefaultParallelism must be positive.")
     .createOptional
 
+  // Lazily get the number of cores to make sure SparkContext created first.
+  private lazy val defaultShufflePartition: Option[Int] = SparkContext.getActive.flatMap { sc =>
+    if (sc.isLocal) Some(SparkContext.numDriverCores(sc.master)) else None
```

Comment:
maybe numDriverCores * 2 or 3? in production practice, we usually expect a larger(2 or 3 times of available exec cores) parallelism of the stage to tackle uneven task runtimes

Reply:
This is only for `local[...]` so I think more common cases are just trying small data out in their local file system -well maybe this isn't the case also. I will need to run some benchmark though.